cupboard_app/queries.py

```diff
@@ -218,13 +273,16 @@ def create_list_ingredient(ingredient: str, amount: int | float, unit: str) -> d
         Exception raised if the ingredient or unit does not exist in the database
         or amount is not int or float type
     """
-    ingredient = Ingredient.objects.get(name=ingredient)
+    try:
+        ingredient_object = CustomIngredient.objects.get(name=ingredient, user=user_id)
```

Comment:
For the filtering of the user here, although this implementation is correct, we should explicitly state the field on the user we are filtering on.
`CustomIngredient.objects.get(name=ingredient, user__id=user_id)`
Specifically **user__id**=user_id